templates/client/html/catalog/detail/body.php

```diff
@@ -48,7 +48,27 @@
 
 
 ?>
-<?php if( isset( $this->detailProductItem ) ) : ?>
+<?php if( isset( $this->detailProductItem ) ) : 
+
+        // preselection of a variant product in a selection article
+        $preSelectVariantAttributes = [];
+        if ($_GET['productId'] && is_numeric($_GET['productId'])):
+
+            if(($listItem = $this->detailProductItem->getListItem( 'product', 'default', (int) $_GET['productId'] ) ) && ( $product = $listItem->getRefItem() ) )
+
+                    $attributes = $product->getRefItems('attribute',null,'variant');
+
+                    foreach($attributes as $attribId => $attribute):
+                        $preSelectVariantAttributes[$attribute->getType()] =  $attribute->getId();
+                    endforeach;
+
+                    if (!empty($preSelectVariantAttributes)):
+                        $preSelectVariantAttributes = htmlspecialchars(json_encode(['attributes' => $preSelectVariantAttributes, 'prodId' => $this->detailProductItem->getId()]), ENT_QUOTES, 'UTF-8');
+                    endif;
```

Comment:
You don't need this, `$enc->attr()` will do the correct encoding and escaping for you